core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -410,6 +423,40 @@ public BlockReader createFileReader(String fileId, long offset, boolean position
         mPageSize);
   }
 
+  @Override
+  public ListenableFuture<List<FileFailure>> load(List<File> files) {
+    List<ListenableFuture<Void>> futures = new ArrayList<>();
+    List<FileFailure> errors = Collections.synchronizedList(new ArrayList<>());
+    for (File file : files) {
+      ListenableFuture<Void> loadFuture = mIOExecutor.submit(() -> {
+        try {
+          load(file.getUfsPath(), file.getMountId(), file.getLength());
+        } catch (Exception e) {
+          AlluxioRuntimeException t = AlluxioRuntimeException.from(e);
+          errors.add(FileFailure.newBuilder().setFile(file).setCode(t.getStatus().getCode().value())
+                                .setMessage(t.getMessage()).build());
+        }
+        return null;
+      });
+      futures.add(loadFuture);
+    }
+    ListenableFuture<List<Void>> future = Futures.successfulAsList(futures);
+    return Futures.transform(future, input -> errors, mIOExecutor);
+  }
+
+  private void load(String ufsPath, long mountId, long length) {
+    Protocol.OpenUfsBlockOptions options =
+        Protocol.OpenUfsBlockOptions.newBuilder().setUfsPath(ufsPath).setMountId(mountId)
+                                    .setNoCache(false).setOffsetInFile(0).setBlockSize(length)
+                                    .build();
+    String fileId = new AlluxioURI(ufsPath).hash();
+    try (BlockReader fileReader = createFileReader(fileId, 0, false, options)) {
+      fileReader.read(0, length);
```

Comment:
actually an unpooled buffer is going to be created out of this length for serving reading purpose, can we have another flavor in pagedreader to only allocate ufsbuf for ufs read?

Reply:
@dbw9580 any suggestion for this? I guess we can wait for the positioned reader and use API: read(long position, byte[] buffer, int length)

Reply:
You can use the `PagedFileReader#transferTo` method and acquire a `PooledDirectNioByteBuf` which is allocated from and recycled by `NioDirectBufferPool`. This should reduce the memory pressure caused by the load operation.

Ideally for such purposes we can have a special implementation of `ByteBuf` that does not copy data and just adjust internal reader and writer indices. This will save even more CPU cycles.

Reply:
@dbw9580 Bowen's position makes sense, and please use a smaller chunk size, e.g. 1MB or 4MB or something.
This is because the new position read impl is direct read, which means sometimes we may allocate some additional buffer in between. e.g. given length is 1GB, we may allocate a 1GB buffer to read from somewhere, and easy to be out of memory.

I am considering how to turn the position read API to be streaming based